### PR TITLE
Changes in SLE 12/15 profiles to support logrotate service

### PR DESCRIPTION
**Pull request by rumch-se**
#### Description:

- _Updates in anssi, pci dss, cis profiles of SLE 12/15 in relation to logrotate service _

#### Rationale:

- At the moment these profiles implement the rule ensure_logrotate_activated, but they do not check if the package is installed and if the timer of this package is enabled.  This PR corrects this. The PR corrects also anssi references into these 3 rules

### CODE REVIEW

**Comment by marcusburghardt** [comment]
@rumch-se , please take a look in the CI tests. A lot is failing. It might be related to the `prodtype` definition in the new rule, since it was set as dependency in another rule used by many products. This is my suspicious but needs to be better investigated.

Another point, I would suggest to not create the dependency relationship in this way. You could define a package platform in the `ensure_logrotate_activated` rule. It would make it simpler and aligned to other similar rules.

**Comment by marcusburghardt** [comment]
> > Hello @marcusburghardt
> > I will provide some additional details about this rule.
> > 1.There is a dependency between the rule ensure_logrotate_activated and the rules package_logrotate_installed and timer_logrotate_enabled. 2.Both rules package_logrotate_installed and timer_logrotate_enabled must be executed before ensure_logrotate_activated 3.For some reason RedHat has CIS numbers in package_logrotate_installed and timer_logrotate_enabled but does not have CCE numbers in identifier sections 4.Usually when we use the old approach for rules definition in the profiles we put the rules in alphabetical order - but in this case ensure_logrotate_activated will be before the rules package_logrotate_installed and timer_logrotate_enabled - i.e. this will impact the order of execution of the rules during the remediation. 5. With the approach via control files we can set the order of the rules to be executed in the more logical way i.e. package_logrotate_installed, timer_logrotate_enabled and logrotate_activated
> > I did a new commit in which I removed prodtype: sle12,sle15 from the rules package_logrotate_installed and timer_logrotate_enabled, but you can see from it, that we have:
> > The following tests FAILED: 140 - missing-cces-rhel7 (Failed) 155 - missing-cces-rhel8 (Failed) 168 - missing-cces-rhel9 (Failed)
> > In this case we have something like a "egg and chicken paradox" - when we have prodtypes sle12,15 - we have errors, but when we don't have them we again have errors, because RedHat had not added CCE codes to 2 rules.
> > I think that some changes should be done on the RedHat site - this issue to be resolved, but I am not in the position to make these changes.
> > Have a nice day Rumen
> 
> @marcusburghardt @rumch-se check if this [086935e](https://github.com/ComplianceAsCode/content/pull/10796/commits/086935e8d671776c079fc009fc0d91a07dd505f5) workaround for failing CCE tests is reasonable from your point of view

@teacup-on-rockingchair would work, but I just reviewed those rules and the CIS benchmarks for RHEL and we can make it simpler. The relevant requirement is `manual` in CIS and therefore we are not including these rules in CIS profiles for RHEL. However, it makes sense to make them available for tailoring files. So I'll quickly add CCEs to these rules and we can rebase this PR.

**Comment by marcusburghardt** [comment]
PR is sent: https://github.com/ComplianceAsCode/content/pull/10904

As soon as it is merged, you can rebase this PR and we should no longer have issues with CCEs.

**Review by teacup-on-rockingchair — APPROVED** [review]
With common of effort  @rumch-se  and @marcusburghardt we managed to pushed this one :fireworks:

**Comment by marcusburghardt** [comment]
There is one required test failing and the error seems legit:
```
TASK [Enable timer logrotate] **************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Could not find the requested service logrotate.timer: host"}
```
I will run some local tests to get more details.

**Comment by teacup-on-rockingchair** [comment]
> Could not find the requested service logrotate.timer

That is strange why it attempts to run that task at the first place on centos. Maybe the problem is with my condition being `{{% if 'sle' in product %}}`, maybe I should convert it to `{{% if product in ["sle15"] %}}` ? what do you think ?

**Comment by marcusburghardt** [comment]
> > Could not find the requested service logrotate.timer
> 
> That is strange why it attempts to run that task at the first place on centos. Maybe the problem is with my condition being `{{% if 'sle' in product %}}`, maybe I should convert it to `{{% if product in ["sle15"] %}}` ? what do you think ?

I can't see any jinja2 condition in this PR.

The issue is caused when trying to remediate the `timer_logrotate_enabled` rule.
This rule was included in the ANSSI control file and consequently is used by RHEL / CentOS and other distros with an ANSSI profile. It is failing because there is no `logrotate.timer` unit in CS8. Checking the `logrotate` Upstream, it seems the `logrotate.timer` unit was introduced since the 3.15 release. In this case, we could include a version in the platform applicability. Could you confirm the logrotate version with `logrotate.timer` unit in `sle12` and `sle15`, please?

**Comment by teacup-on-rockingchair** [comment]
> > > Could not find the requested service logrotate.timer
> > 
> > 
> > That is strange why it attempts to run that task at the first place on centos. Maybe the problem is with my condition being `{{% if 'sle' in product %}}`, maybe I should convert it to `{{% if product in ["sle15"] %}}` ? what do you think ?
> 
> I can't see any jinja2 condition in this PR.
> 
I thought that the error is in ./linux_os/guide/system/logging/log_rotation/ensure_logrotate_activated/ansible/shared.yml, where there is a a check if product is `sle` or not.

> The issue is caused when trying to remediate the `timer_logrotate_enabled` rule. This rule was included in the ANSSI control file and consequently is used by RHEL / CentOS and other distros with an ANSSI profile. It is failing because there is no `logrotate.timer` unit in CS8. Checking the `logrotate` Upstream, it seems the `logrotate.timer` unit was introduced since the 3.15 release. In this case, we could include a version in the platform applicability. Could you confirm the logrotate version with `logrotate.timer` unit in `sle12` and `sle15`, please?

For sure in SLE12-SP5 - we have the package version 3.11.0, and the timer defined, so putting a version 3.14 will make the rule there inapplicable, while it can be used.
 Also it looks kind of dangerous to assume same package versions on different distros, if I understand correctly the suggested change, can we add also OS as platform limitation or do some extra code in the oval to report not applicable?

**Comment by teacup-on-rockingchair** [comment]
> > while it can be used
> 
> Ok. In this case the suggested update to the platform is not good. Maybe for now we have to include the `prodtype: sle12,sle15,rhel9` in this rule or even something like `os_linux[rhel]>=9.0 and os_linux[sle] ...` Other products can be included after, on demand. What do you think?

Added `prodtype` clause to the rule definition,  and also jinja condition in the `ensure_logrotate_activated` oval, so the rule is applicable only when we have logrotate.timer definition. @ComplianceAsCode/oracle-maintainers , @ComplianceAsCode/ubuntu-maintainers , can you please review the last change and suggest products, which should be added to the `prodtype` clause?

File: linux_os/guide/system/logging/log_rotation/timer_logrotate_enabled/rule.yml
Comment (marcusburghardt) [inline linux_os/guide/system/logging/log_rotation/timer_logrotate_enabled/rule.yml:3]:
```suggestion
prodtype: rhel9,sle12,sle15
```

File: linux_os/guide/system/logging/log_rotation/ensure_logrotate_activated/oval/shared.xml
Comment (marcusburghardt) [inline linux_os/guide/system/logging/log_rotation/ensure_logrotate_activated/oval/shared.xml:14]:
```suggestion
{{% if product in ["rhel9", "sle12", "sle15"] %}}
```

**Review by marcusburghardt — CHANGES_REQUESTED** [review]

File: linux_os/guide/system/logging/log_rotation/timer_logrotate_enabled/rule.yml
Comment (freddieRv) [reply linux_os/guide/system/logging/log_rotation/timer_logrotate_enabled/rule.yml:3]: `ol9` can also be added to the `prodtype` list. It is the only OL release which currently supports the `logrotate.timer` service.

Comment (freddieRv) [reply linux_os/guide/system/logging/log_rotation/timer_logrotate_enabled/rule.yml:3]: Also worth mentioning: [Release notes](https://github.com/logrotate/logrotate/releases/tag/3.11.0) for logrotate's release 3.11.0 mentions:
> _Import systemd service and timer for logrotate from openSUSE._

So this feature might be available from this release, and not starting from 3.14.0 as suggested in some comment above.

**Review by marcusburghardt — APPROVED** [review]
Thanks

File: linux_os/guide/system/logging/log_rotation/ensure_logrotate_activated/oval/shared.xml
Comment (yuumasato) [inline linux_os/guide/system/logging/log_rotation/ensure_logrotate_activated/oval/shared.xml:7]: I wonder if this `extended_definition` is necessary when the rule has `platform: package[logrotate]`

**Pull request merged**
